src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1309,6 +1325,19 @@ private void ImportInitObj(int token)
 
         private void ImportBox(int token)
         {
+            TypeDesc type = ResolveTypeToken(token);
+
+            if (type.IsValueType)
+            {
+                if (type.IsNullable)
+                    throw new NotImplementedException();
+
+                var value = _stack.Pop();
+
+                var boxPtr = LLVM.BuildMalloc(_builder, GetLLVMTypeForTypeDesc(type), string.Empty);
+                LLVM.BuildStore(_builder, value.LLVMValue, boxPtr);
```

Comment:
This needs to do more than store the value. Take a look at https://github.com/dotnet/corert/blob/2b40a707239a9974a63cf992ce6a6f8303f5b362/src/Native/Runtime/gcrhenv.cpp#L763 for what the runtime does. You can skip the write barrier code, but I think leaving room for the object header and padding is important. We might also need the eetypeptr. @davidwrighton , do we have that available in some way?

Reply:
I can easily add padding but it looks like eetypeptr is the same thing as MethodTable* in cppcodegen. This looks like it will be a somewhat large change to generate. I'll try my hand at outputting equivalent data from LLVM, but I would prefer to separate it from this PR. So I'd like to just let this PR sit until that one is done.

Reply:
That sounds reasonable. I believe we're already generating EETypes via WebAssemblyObjectWriter, so hopefully we can get the pointers without too much trouble. @davidwrighton  or @MichalStrehovsky , can you chime in on details? (That will help with newobj and newarr too)

Reply:
Ah, I think i found it in WebAssemblyObjectWriter.ObjectNodeDataEmission, that will make things much easier.